client_integration_test.go

```diff
@@ -2466,7 +2498,9 @@ func benchmarkWriteTo(b *testing.B, bufsize int, delay time.Duration) {
 	b.ResetTimer()
 	b.SetBytes(int64(size))
 
-	buf := new(bytes.Buffer)
+	buf := &writeToBuffer{
+		b: make([]byte, 0, size),
+	}
```

Comment:
This should go before the `b.ResetTimer()` so that it is not included in the memory benchmark stats.